### PR TITLE
Add error helpers

### PR DESCRIPTION
**Pull request by Ntr0**
To improve the sdk's usability, some helper functions were added to check if an error is of given type, e.g. IsStatusNotFound(error) returns true if the api returned 404 for the request.

### CODE REVIEW

File: error.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+func IsHttpStatus(err error, status int) bool {
+	if err, ok := err.(ApiError); ok {
+		return err.HttpStatusCode() == status
+	}
+	return false
+}
+
+// IsStatusOK - (200)
+func IsStatusOK(err error) bool {
```
[the remark this answers is not in the document]
Comment (Ntr0) [reply error.go]: Take a look at https://github.com/profitbricks/profitbricks-sdk-go/blob/1d2db5f00bf5dd0b6c29273541c71c60cdf4d4d4/req.go#L154 : An error is returned, if the repsonse status code is not the expected. So I simply added all errors that ionos documented as possible http status codes.

File: error.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	return IsHttpStatus(err, http.StatusForbidden)
+}
+
+// IsStatusNotFound - (404) if resource does not exist
+func IsStatusNotFound(err error) bool {
+	return IsHttpStatus(err, http.StatusNotFound)
+}
+
+// IsStatusMethodNotAllowed - (405) Use for any POST, PUT, PATCH, or DELETE performed
+// on read-only resources. This is also there response to PATCH requests
```
[the remark this answers is not in the document]
Comment (Ntr0) [reply error.go]: In general yes, but on the other hand I just took the doc from ionos regarding this codes.

[the remark this answers is not in the document]
Comment (Ntr0) [reply error.go]: That's also where they typo originates :(

File: error.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (piepmatz) [reply error.go]: fair enough

**Pull request merged**